File: bindings/typescript/examples/hyperliquid-provider.mjs
// Read Hyperliquid public data and optional address-scoped Info data.
import { Exchange, HyperliquidAdapter, Market, initialize } from "@jabdori/maxt/node";

await initialize();

const market = Market.perpetual(Exchange.Hyperliquid, "BTC", "USDC");
const adapter = new HyperliquidAdapter();
const [mids, book, trades] = await Promise.all([
  adapter.allMids(),
  adapter.l2Book(market),
  adapter.recentTrades(market),
]);
console.log(`${mids.length} mid prices; ${book.bids.length + book.asks.length} levels; ${trades.length} trades`);

const address = process.env.HYPERLIQUID_ADDRESS;
if (address === undefined) {
  console.log("Set HYPERLIQUID_ADDRESS for address-scoped Info reads; a private key is not needed.");
} else {
  const orders = await new HyperliquidAdapter({ address }).basicOpenOrders();
  console.log(`${orders.length} address-scoped open orders`);
}

File: bindings/python/python/maxt/examples/hyperliquid_provider.py
"""Read Hyperliquid public data and optional address-scoped Info data."""

from __future__ import annotations

import asyncio
import os

from maxt import Exchange, HyperliquidAdapter, Market


async def main() -> None:
    market = Market.perpetual(Exchange.HYPERLIQUID, "BTC", "USDC")
    adapter = HyperliquidAdapter()
    mids = await adapter.all_mids()
    book = await adapter.l2_book(market)
    trades = await adapter.recent_trades(market)
    print(f"{len(mids)} mid prices; {len(book.bids) + len(book.asks)} levels; {len(trades)} trades")

    address = os.getenv("HYPERLIQUID_ADDRESS")
    if address:
        orders = await HyperliquidAdapter(address=address).basic_open_orders()
        print(f"{len(orders)} address-scoped open orders")
    else:
        print("set HYPERLIQUID_ADDRESS for address-scoped Info reads; a private key is not needed")


if __name__ == "__main__":
    asyncio.run(main())
